core/include/xfa.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_util
+ * @brief       Cross File Arrays
+ * @{
+ *
+ * This macro, in combination with an entry in the linker scripts, allows the
+ * definition of constant arrays to be spread over multiple C compilation
+ * units. These arrays are called "cross-file arrays" or short xfa.
+ *
+ *
+ *
+ * @file
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef XFA_H
+#define XFA_H
+
+#if defined(__arm__)
+#define XFA_EXTRA volatile
+#else
+#define XFA_EXTRA
+#endif
+
+/**
+ * @brief helper macro for other XFA_* macros
+ *
+ * @internal
+ */
+#define _XFA(type, name, prio) __attribute__((used)) \
+    __attribute__((section(".xfa." #name "." #prio))) \
+    XFA_EXTRA \
+    __attribute__((aligned(__alignof__(type))))
```

Comment:
I still wonder if the alignment is necessary or if the compiler will do this anyway. But seems to me it is not quite trivial to write a valid test app for this?!

Reply:
Honestly, I don't remember. I did have oddly aligned data at some point while testing. I'll remove that for now.